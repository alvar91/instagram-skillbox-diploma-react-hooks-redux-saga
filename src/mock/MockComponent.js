import React from "react";

function MockComponent() {
  return <h2>Mock Component</h2>;
}

export default MockComponent;
