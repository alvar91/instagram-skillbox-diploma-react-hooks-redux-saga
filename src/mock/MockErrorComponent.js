import React from "react";

function MockErrorComponent() {
  React.useEffect(() => {
    throw new Error("Mock error");
  }, []);

  return <h2>Mock error</h2>;
}

export default MockErrorComponent;
