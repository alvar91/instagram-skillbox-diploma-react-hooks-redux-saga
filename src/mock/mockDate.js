export const mockDate = "2021-02-11T17:25:19-05:00";
export const correctMockDate = "Fri, Feb 12, 2021";
