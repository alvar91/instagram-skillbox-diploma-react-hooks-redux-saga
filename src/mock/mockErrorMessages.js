export const mockAuthErrorMessage =
  "Failed to get access token. Please try to log in again";
export const mockErrorMeta = "meta error";
