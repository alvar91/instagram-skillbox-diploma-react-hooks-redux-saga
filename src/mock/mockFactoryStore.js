/**
 * Function to create a mock store with input options
 * @param {object} storeOptionsObj contains options to create the mock store
 * @returns {object} mock store
 */
export const mockFactoryStore = ({postsObj = {}, usersObj = {}, authObj = {}}) => {
  const {
    posts = [],
    currentPage = 1,
    isFetchingPost = false,
    hasErrorPosts = false,
    errorMessagePosts = "",
  } = postsObj;

  const { users = [] } = usersObj;

  const {
    isFetchingAuth = false,
    isLoggedIn = false,
    profile = [],
    hasErrorAuth = false,
    errorMessageAuth = "",
  } = authObj;

  return {
    /* posts state */
    posts: {
      isFetchingPost: isFetchingPost,
      currentPage: currentPage,
      posts: posts,
      users: users,
      hasError: hasErrorPosts,
      errorMessage: errorMessagePosts,
    },
    /* auth state */
    auth: {
      isFetchingAuth: isFetchingAuth,
      isLoggedIn: isLoggedIn,
      profile: profile,
      hasError: hasErrorAuth,
      errorMessage: errorMessageAuth,
    },
  };
};
