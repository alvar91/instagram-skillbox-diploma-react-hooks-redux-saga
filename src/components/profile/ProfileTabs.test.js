// Core
import React from "react";
import { configure, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import ProfileTabs from "./ProfileTabs";
import { mockProfile } from "../../mock/mockProfile";

configure({ adapter: new Adapter() });

/**
 * Setup function for ProfileTabs component
 * @param {object} storeOptionsObj contains options for creating the mock store
 * @returns {ReactRenderer}
 */
const setupTestRenderer = () => {
  return mount(<ProfileTabs user={mockProfile} isOwner />);
};

describe("ProfileTabs", () => {
  describe("ProfileTabs with mock post state", () => {
    let testRenderer;
    let component;
    const props = { user: mockProfile, isOwner: true };

    beforeAll(() => {
      testRenderer = setupTestRenderer();
      component = testRenderer.find(ProfileTabs);
    });

    it("ProfileTabs should correctly renders markup", () => {
      expect(shallowToJson(testRenderer)).toMatchSnapshot();
    });

    it("ProfileTabs should contain correct props", () => {
      expect(component).toHaveLength(1);
      expect(component.props()).toMatchObject(props);
    });
  });
});
