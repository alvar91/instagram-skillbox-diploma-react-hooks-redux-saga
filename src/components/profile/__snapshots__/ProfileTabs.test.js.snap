// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileTabs ProfileTabs with mock post state ProfileTabs should correctly renders markup 1`] = `
<ProfileTabs
  isOwner={true}
  user={
    Object {
      "accepted_tos": true,
      "allow_messages": true,
      "badge": null,
      "bio": "Code massacre",
      "confirmed": true,
      "dmca_verification": "requested",
      "downloads": 0,
      "email": "aleksey91scorp@bk.ru",
      "first_name": "Aleksey",
      "followed_by_user": false,
      "followers_count": 0,
      "following_count": 0,
      "for_hire": false,
      "id": "WWikOqpDkoQ",
      "instagram_username": null,
      "last_name": "Varov",
      "links": Object {
        "followers": "https://api.unsplash.com/users/alvar91/followers",
        "following": "https://api.unsplash.com/users/alvar91/following",
        "html": "https://unsplash.com/@alvar91",
        "likes": "https://api.unsplash.com/users/alvar91/likes",
        "photos": "https://api.unsplash.com/users/alvar91/photos",
        "portfolio": "https://api.unsplash.com/users/alvar91/portfolio",
        "self": "https://api.unsplash.com/users/alvar91",
      },
      "location": "Moscow",
      "meta": Object {
        "index": false,
      },
      "name": "Aleksey Varov",
      "numeric_id": 5337716,
      "photos": Array [],
      "portfolio_url": "https://github.com/alvar91",
      "profile_image": Object {
        "large": "https://images.unsplash.com/profile-1610144463792-69ea2bb87f2bimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
        "medium": "https://images.unsplash.com/profile-1610144463792-69ea2bb87f2bimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        "small": "https://images.unsplash.com/profile-1610144463792-69ea2bb87f2bimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
      },
      "tags": Object {
        "aggregated": Array [
          Object {
            "title": "building",
            "type": "search",
          },
          Object {
            "title": "silhouette",
            "type": "search",
          },
          Object {
            "source": Object {
              "ancestry": Object {
                "category": Object {
                  "pretty_slug": "Nature",
                  "slug": "nature",
                },
                "subcategory": Object {
                  "pretty_slug": "Landscape",
                  "slug": "landscape",
                },
                "type": Object {
                  "pretty_slug": "Images",
                  "slug": "images",
                },
              },
              "cover_photo": Object {
                "alt_description": "white wooden tree surround by grass field during sunset",
                "blur_hash": "LLDJCiNz0M%0.AkDNHxaA1WX%1Rl",
                "categories": Array [],
                "color": "#59738c",
                "created_at": "2015-06-05T00:06:06-04:00",
                "current_user_collections": Array [],
                "description": "Mountain Valley",
                "height": 3026,
                "id": "LJD6U920zVo",
                "liked_by_user": false,
                "likes": 512,
                "links": Object {
                  "download": "https://unsplash.com/photos/LJD6U920zVo/download",
                  "download_location": "https://api.unsplash.com/photos/LJD6U920zVo/download",
                  "html": "https://unsplash.com/photos/LJD6U920zVo",
                  "self": "https://api.unsplash.com/photos/LJD6U920zVo",
                },
                "promoted_at": "2015-06-05T00:06:06-04:00",
                "sponsorship": null,
                "updated_at": "2021-02-10T21:00:13-05:00",
                "urls": Object {
                  "full": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb",
                  "raw": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1",
                  "regular": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max",
                  "small": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max",
                  "thumb": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max",
                },
                "user": Object {
                  "accepted_tos": false,
                  "bio": "Jasper is a Dutch photographer and graphic designer with a passion for landscapes, based in beautiful Raglan, New Zealand 🏄
 Framed and unframed prints of his work, as well as fine art prints are available from his website artprints.co.nz ",
                  "first_name": "Jasper",
                  "id": "4M-5qUF81Ks",
                  "instagram_username": "photoadventurenz",
                  "last_name": "Boer",
                  "links": Object {
                    "followers": "https://api.unsplash.com/users/jasperboer/followers",
                    "following": "https://api.unsplash.com/users/jasperboer/following",
                    "html": "https://unsplash.com/@jasperboer",
                    "likes": "https://api.unsplash.com/users/jasperboer/likes",
                    "photos": "https://api.unsplash.com/users/jasperboer/photos",
                    "portfolio": "https://api.unsplash.com/users/jasperboer/portfolio",
                    "self": "https://api.unsplash.com/users/jasperboer",
                  },
                  "location": "Raglan, New Zealand",
                  "name": "Jasper Boer",
                  "portfolio_url": "http://http//artprints.co.nz",
                  "profile_image": Object {
                    "large": "https://images.unsplash.com/profile-1456089309179-41943ec63a94?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
                    "medium": "https://images.unsplash.com/profile-1456089309179-41943ec63a94?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
                    "small": "https://images.unsplash.com/profile-1456089309179-41943ec63a94?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
                  },
                  "total_collections": 0,
                  "total_likes": 0,
                  "total_photos": 8,
                  "twitter_username": "jasperboer",
                  "updated_at": "2021-02-11T08:42:51-05:00",
                  "username": "jasperboer",
                },
                "width": 4797,
              },
              "description": "Choose from a curated selection of landscape photos. Always free on Unsplash.",
              "meta_description": "Choose from hundreds of free landscape pictures. Download HD landscape photos for free on Unsplash.",
              "meta_title": "Stunning Landscape Pictures | Download Free Images on Unsplash",
              "subtitle": "Download free landscape images",
              "title": "Landscape Images & Pictures",
            },
            "title": "landscape",
            "type": "landing_page",
          },
          Object {
            "title": "outdoor",
            "type": "search",
          },
          Object {
            "title": "mercede",
            "type": "search",
          },
          Object {
            "title": "racing",
            "type": "search",
          },
          Object {
            "title": "pit lane",
            "type": "search",
          },
          Object {
            "title": "race car",
            "type": "search",
          },
          Object {
            "title": "grand prix",
            "type": "search",
          },
          Object {
            "title": "spa francorchamp",
            "type": "search",
          },
          Object {
            "title": "24h",
            "type": "search",
          },
          Object {
            "title": "mercedes benz",
            "type": "search",
          },
          Object {
            "title": "amg",
            "type": "search",
          },
          Object {
            "title": "fast",
            "type": "search",
          },
          Object {
            "title": "belgium",
            "type": "search",
          },
          Object {
            "title": "stavelot",
            "type": "search",
          },
          Object {
            "title": "route du circuit",
            "type": "search",
          },
          Object {
            "title": "spa-francorchamp",
            "type": "search",
          },
          Object {
            "title": "wheel",
            "type": "search",
          },
          Object {
            "title": "vehicle",
            "type": "search",
          },
          Object {
            "title": "transportation",
            "type": "search",
          },
          Object {
            "title": "tire",
            "type": "search",
          },
          Object {
            "title": "sports car",
            "type": "search",
          },
          Object {
            "title": "spoke",
            "type": "search",
          },
          Object {
            "title": "officer",
            "type": "search",
          },
          Object {
            "title": "military uniform",
            "type": "search",
          },
          Object {
            "title": "military",
            "type": "search",
          },
          Object {
            "title": "machine",
            "type": "search",
          },
          Object {
            "title": "coupe",
            "type": "search",
          },
          Object {
            "title": "car wheel",
            "type": "search",
          },
        ],
        "custom": Array [
          Object {
            "source": Object {
              "ancestry": Object {
                "category": Object {
                  "pretty_slug": "Nature",
                  "slug": "nature",
                },
                "type": Object {
                  "pretty_slug": "Images",
                  "slug": "images",
                },
              },
              "cover_photo": Object {
                "alt_description": "star in space",
                "blur_hash": "L21o;CogI7WARNaxt9j]Mvaxxwof",
                "categories": Array [],
                "color": "#262640",
                "created_at": "2018-10-15T04:58:20-04:00",
                "current_user_collections": Array [],
                "description": "lost in the sky",
                "height": 3480,
                "id": "VE5FRc7uiC4",
                "liked_by_user": false,
                "likes": 107,
                "links": Object {
                  "download": "https://unsplash.com/photos/VE5FRc7uiC4/download",
                  "download_location": "https://api.unsplash.com/photos/VE5FRc7uiC4/download",
                  "html": "https://unsplash.com/photos/VE5FRc7uiC4",
                  "self": "https://api.unsplash.com/photos/VE5FRc7uiC4",
                },
                "promoted_at": "2018-10-15T08:23:00-04:00",
                "sponsorship": null,
                "updated_at": "2021-01-21T11:04:40-05:00",
                "urls": Object {
                  "full": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb",
                  "raw": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1",
                  "regular": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max",
                  "small": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max",
                  "thumb": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max",
                },
                "user": Object {
                  "accepted_tos": true,
                  "bio": "Create The Moment",
                  "first_name": "Akin",
                  "id": "PvaYY7qgvqU",
                  "instagram_username": "akincakiner",
                  "last_name": null,
                  "links": Object {
                    "followers": "https://api.unsplash.com/users/akin/followers",
                    "following": "https://api.unsplash.com/users/akin/following",
                    "html": "https://unsplash.com/@akin",
                    "likes": "https://api.unsplash.com/users/akin/likes",
                    "photos": "https://api.unsplash.com/users/akin/photos",
                    "portfolio": "https://api.unsplash.com/users/akin/portfolio",
                    "self": "https://api.unsplash.com/users/akin",
                  },
                  "location": "almelo",
                  "name": "Akin",
                  "portfolio_url": "https://akincakiner.com/",
                  "profile_image": Object {
                    "large": "https://images.unsplash.com/profile-1578436703762-5a9ab2f10de8image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
                    "medium": "https://images.unsplash.com/profile-1578436703762-5a9ab2f10de8image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
                    "small": "https://images.unsplash.com/profile-1578436703762-5a9ab2f10de8image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
                  },
                  "total_collections": 0,
                  "total_likes": 56,
                  "total_photos": 276,
                  "twitter_username": "pausyworld",
                  "updated_at": "2021-01-22T02:37:38-05:00",
                  "username": "akin",
                },
                "width": 4640,
              },
              "description": "Nature produces the most astoundingly beautiful images: the swirling lava of a volcano, palm trees against a blue sky, snow-capped mountains towering above. Unsplash has magnificent , high-quality photos of all the delights that nature has to offer.",
              "meta_description": "Choose from hundreds of free nature pictures. Download HD nature photos for free on Unsplash.",
              "meta_title": "100+ Nature Pictures | Download Free Images & Stock Photos on Unsplash",
              "subtitle": "Download free nature images",
              "title": "Nature Images",
            },
            "title": "nature",
            "type": "landing_page",
          },
          Object {
            "title": "code",
            "type": "search",
          },
          Object {
            "title": "javascript",
            "type": "search",
          },
          Object {
            "title": "typescript",
            "type": "search",
          },
          Object {
            "title": "web",
            "type": "search",
          },
        ],
      },
      "total_collections": 0,
      "total_likes": 17,
      "total_photos": 1,
      "twitter_username": null,
      "uid": "WWikOqpDkoQ",
      "unlimited_uploads": false,
      "updated_at": "2021-02-14T07:31:42-05:00",
      "uploads_remaining": 10,
      "username": "alvar91",
    }
  }
>
  <section
    className="makeStyles-section-2"
  >
    <Hidden
      xsDown={true}
    >
      <WithWidth(HiddenJs)
        lgDown={false}
        lgUp={false}
        mdDown={false}
        mdUp={false}
        smDown={false}
        smUp={false}
        xlDown={false}
        xlUp={false}
        xsDown={true}
        xsUp={false}
      />
    </Hidden>
    <Hidden
      xsDown={true}
    >
      <WithWidth(HiddenJs)
        lgDown={false}
        lgUp={false}
        mdDown={false}
        mdUp={false}
        smDown={false}
        smUp={false}
        xlDown={false}
        xlUp={false}
        xsDown={true}
        xsUp={false}
      />
    </Hidden>
    <Hidden
      smUp={true}
    >
      <WithWidth(HiddenJs)
        lgDown={false}
        lgUp={false}
        mdDown={false}
        mdUp={false}
        smDown={false}
        smUp={true}
        xlDown={false}
        xlUp={false}
        xsDown={false}
        xsUp={false}
      />
    </Hidden>
    <Hidden
      smUp={true}
    >
      <WithWidth(HiddenJs)
        lgDown={false}
        lgUp={false}
        mdDown={false}
        mdUp={false}
        smDown={false}
        smUp={true}
        xlDown={false}
        xlUp={false}
        xsDown={false}
        xsUp={false}
      />
    </Hidden>
    <ProfilePosts
      isOwner={true}
      user={
        Object {
          "accepted_tos": true,
          "allow_messages": true,
          "badge": null,
          "bio": "Code massacre",
          "confirmed": true,
          "dmca_verification": "requested",
          "downloads": 0,
          "email": "aleksey91scorp@bk.ru",
          "first_name": "Aleksey",
          "followed_by_user": false,
          "followers_count": 0,
          "following_count": 0,
          "for_hire": false,
          "id": "WWikOqpDkoQ",
          "instagram_username": null,
          "last_name": "Varov",
          "links": Object {
            "followers": "https://api.unsplash.com/users/alvar91/followers",
            "following": "https://api.unsplash.com/users/alvar91/following",
            "html": "https://unsplash.com/@alvar91",
            "likes": "https://api.unsplash.com/users/alvar91/likes",
            "photos": "https://api.unsplash.com/users/alvar91/photos",
            "portfolio": "https://api.unsplash.com/users/alvar91/portfolio",
            "self": "https://api.unsplash.com/users/alvar91",
          },
          "location": "Moscow",
          "meta": Object {
            "index": false,
          },
          "name": "Aleksey Varov",
          "numeric_id": 5337716,
          "photos": Array [],
          "portfolio_url": "https://github.com/alvar91",
          "profile_image": Object {
            "large": "https://images.unsplash.com/profile-1610144463792-69ea2bb87f2bimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
            "medium": "https://images.unsplash.com/profile-1610144463792-69ea2bb87f2bimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
            "small": "https://images.unsplash.com/profile-1610144463792-69ea2bb87f2bimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
          },
          "tags": Object {
            "aggregated": Array [
              Object {
                "title": "building",
                "type": "search",
              },
              Object {
                "title": "silhouette",
                "type": "search",
              },
              Object {
                "source": Object {
                  "ancestry": Object {
                    "category": Object {
                      "pretty_slug": "Nature",
                      "slug": "nature",
                    },
                    "subcategory": Object {
                      "pretty_slug": "Landscape",
                      "slug": "landscape",
                    },
                    "type": Object {
                      "pretty_slug": "Images",
                      "slug": "images",
                    },
                  },
                  "cover_photo": Object {
                    "alt_description": "white wooden tree surround by grass field during sunset",
                    "blur_hash": "LLDJCiNz0M%0.AkDNHxaA1WX%1Rl",
                    "categories": Array [],
                    "color": "#59738c",
                    "created_at": "2015-06-05T00:06:06-04:00",
                    "current_user_collections": Array [],
                    "description": "Mountain Valley",
                    "height": 3026,
                    "id": "LJD6U920zVo",
                    "liked_by_user": false,
                    "likes": 512,
                    "links": Object {
                      "download": "https://unsplash.com/photos/LJD6U920zVo/download",
                      "download_location": "https://api.unsplash.com/photos/LJD6U920zVo/download",
                      "html": "https://unsplash.com/photos/LJD6U920zVo",
                      "self": "https://api.unsplash.com/photos/LJD6U920zVo",
                    },
                    "promoted_at": "2015-06-05T00:06:06-04:00",
                    "sponsorship": null,
                    "updated_at": "2021-02-10T21:00:13-05:00",
                    "urls": Object {
                      "full": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb",
                      "raw": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1",
                      "regular": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max",
                      "small": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max",
                      "thumb": "https://images.unsplash.com/photo-1433477155337-9aea4e790195?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max",
                    },
                    "user": Object {
                      "accepted_tos": false,
                      "bio": "Jasper is a Dutch photographer and graphic designer with a passion for landscapes, based in beautiful Raglan, New Zealand 🏄
 Framed and unframed prints of his work, as well as fine art prints are available from his website artprints.co.nz ",
                      "first_name": "Jasper",
                      "id": "4M-5qUF81Ks",
                      "instagram_username": "photoadventurenz",
                      "last_name": "Boer",
                      "links": Object {
                        "followers": "https://api.unsplash.com/users/jasperboer/followers",
                        "following": "https://api.unsplash.com/users/jasperboer/following",
                        "html": "https://unsplash.com/@jasperboer",
                        "likes": "https://api.unsplash.com/users/jasperboer/likes",
                        "photos": "https://api.unsplash.com/users/jasperboer/photos",
                        "portfolio": "https://api.unsplash.com/users/jasperboer/portfolio",
                        "self": "https://api.unsplash.com/users/jasperboer",
                      },
                      "location": "Raglan, New Zealand",
                      "name": "Jasper Boer",
                      "portfolio_url": "http://http//artprints.co.nz",
                      "profile_image": Object {
                        "large": "https://images.unsplash.com/profile-1456089309179-41943ec63a94?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
                        "medium": "https://images.unsplash.com/profile-1456089309179-41943ec63a94?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
                        "small": "https://images.unsplash.com/profile-1456089309179-41943ec63a94?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
                      },
                      "total_collections": 0,
                      "total_likes": 0,
                      "total_photos": 8,
                      "twitter_username": "jasperboer",
                      "updated_at": "2021-02-11T08:42:51-05:00",
                      "username": "jasperboer",
                    },
                    "width": 4797,
                  },
                  "description": "Choose from a curated selection of landscape photos. Always free on Unsplash.",
                  "meta_description": "Choose from hundreds of free landscape pictures. Download HD landscape photos for free on Unsplash.",
                  "meta_title": "Stunning Landscape Pictures | Download Free Images on Unsplash",
                  "subtitle": "Download free landscape images",
                  "title": "Landscape Images & Pictures",
                },
                "title": "landscape",
                "type": "landing_page",
              },
              Object {
                "title": "outdoor",
                "type": "search",
              },
              Object {
                "title": "mercede",
                "type": "search",
              },
              Object {
                "title": "racing",
                "type": "search",
              },
              Object {
                "title": "pit lane",
                "type": "search",
              },
              Object {
                "title": "race car",
                "type": "search",
              },
              Object {
                "title": "grand prix",
                "type": "search",
              },
              Object {
                "title": "spa francorchamp",
                "type": "search",
              },
              Object {
                "title": "24h",
                "type": "search",
              },
              Object {
                "title": "mercedes benz",
                "type": "search",
              },
              Object {
                "title": "amg",
                "type": "search",
              },
              Object {
                "title": "fast",
                "type": "search",
              },
              Object {
                "title": "belgium",
                "type": "search",
              },
              Object {
                "title": "stavelot",
                "type": "search",
              },
              Object {
                "title": "route du circuit",
                "type": "search",
              },
              Object {
                "title": "spa-francorchamp",
                "type": "search",
              },
              Object {
                "title": "wheel",
                "type": "search",
              },
              Object {
                "title": "vehicle",
                "type": "search",
              },
              Object {
                "title": "transportation",
                "type": "search",
              },
              Object {
                "title": "tire",
                "type": "search",
              },
              Object {
                "title": "sports car",
                "type": "search",
              },
              Object {
                "title": "spoke",
                "type": "search",
              },
              Object {
                "title": "officer",
                "type": "search",
              },
              Object {
                "title": "military uniform",
                "type": "search",
              },
              Object {
                "title": "military",
                "type": "search",
              },
              Object {
                "title": "machine",
                "type": "search",
              },
              Object {
                "title": "coupe",
                "type": "search",
              },
              Object {
                "title": "car wheel",
                "type": "search",
              },
            ],
            "custom": Array [
              Object {
                "source": Object {
                  "ancestry": Object {
                    "category": Object {
                      "pretty_slug": "Nature",
                      "slug": "nature",
                    },
                    "type": Object {
                      "pretty_slug": "Images",
                      "slug": "images",
                    },
                  },
                  "cover_photo": Object {
                    "alt_description": "star in space",
                    "blur_hash": "L21o;CogI7WARNaxt9j]Mvaxxwof",
                    "categories": Array [],
                    "color": "#262640",
                    "created_at": "2018-10-15T04:58:20-04:00",
                    "current_user_collections": Array [],
                    "description": "lost in the sky",
                    "height": 3480,
                    "id": "VE5FRc7uiC4",
                    "liked_by_user": false,
                    "likes": 107,
                    "links": Object {
                      "download": "https://unsplash.com/photos/VE5FRc7uiC4/download",
                      "download_location": "https://api.unsplash.com/photos/VE5FRc7uiC4/download",
                      "html": "https://unsplash.com/photos/VE5FRc7uiC4",
                      "self": "https://api.unsplash.com/photos/VE5FRc7uiC4",
                    },
                    "promoted_at": "2018-10-15T08:23:00-04:00",
                    "sponsorship": null,
                    "updated_at": "2021-01-21T11:04:40-05:00",
                    "urls": Object {
                      "full": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb",
                      "raw": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1",
                      "regular": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max",
                      "small": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max",
                      "thumb": "https://images.unsplash.com/photo-1539593395743-7da5ee10ff07?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max",
                    },
                    "user": Object {
                      "accepted_tos": true,
                      "bio": "Create The Moment",
                      "first_name": "Akin",
                      "id": "PvaYY7qgvqU",
                      "instagram_username": "akincakiner",
                      "last_name": null,
                      "links": Object {
                        "followers": "https://api.unsplash.com/users/akin/followers",
                        "following": "https://api.unsplash.com/users/akin/following",
                        "html": "https://unsplash.com/@akin",
                        "likes": "https://api.unsplash.com/users/akin/likes",
                        "photos": "https://api.unsplash.com/users/akin/photos",
                        "portfolio": "https://api.unsplash.com/users/akin/portfolio",
                        "self": "https://api.unsplash.com/users/akin",
                      },
                      "location": "almelo",
                      "name": "Akin",
                      "portfolio_url": "https://akincakiner.com/",
                      "profile_image": Object {
                        "large": "https://images.unsplash.com/profile-1578436703762-5a9ab2f10de8image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
                        "medium": "https://images.unsplash.com/profile-1578436703762-5a9ab2f10de8image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
                        "small": "https://images.unsplash.com/profile-1578436703762-5a9ab2f10de8image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
                      },
                      "total_collections": 0,
                      "total_likes": 56,
                      "total_photos": 276,
                      "twitter_username": "pausyworld",
                      "updated_at": "2021-01-22T02:37:38-05:00",
                      "username": "akin",
                    },
                    "width": 4640,
                  },
                  "description": "Nature produces the most astoundingly beautiful images: the swirling lava of a volcano, palm trees against a blue sky, snow-capped mountains towering above. Unsplash has magnificent , high-quality photos of all the delights that nature has to offer.",
                  "meta_description": "Choose from hundreds of free nature pictures. Download HD nature photos for free on Unsplash.",
                  "meta_title": "100+ Nature Pictures | Download Free Images & Stock Photos on Unsplash",
                  "subtitle": "Download free nature images",
                  "title": "Nature Images",
                },
                "title": "nature",
                "type": "landing_page",
              },
              Object {
                "title": "code",
                "type": "search",
              },
              Object {
                "title": "javascript",
                "type": "search",
              },
              Object {
                "title": "typescript",
                "type": "search",
              },
              Object {
                "title": "web",
                "type": "search",
              },
            ],
          },
          "total_collections": 0,
          "total_likes": 17,
          "total_photos": 1,
          "twitter_username": null,
          "uid": "WWikOqpDkoQ",
          "unlimited_uploads": false,
          "updated_at": "2021-02-14T07:31:42-05:00",
          "uploads_remaining": 10,
          "username": "alvar91",
        }
      }
    >
      <section
        className="makeStyles-profilePostsSection-22"
      >
        <div
          className="makeStyles-noContent-14"
        >
          <div
            className="makeStyles-uploadPhotoIcon-24"
          />
          <WithStyles(ForwardRef(Typography))
            variant="h4"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              variant="h4"
            >
              <h4
                className="MuiTypography-root MuiTypography-h4"
              >
                Upload a Photo
              </h4>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </div>
      </section>
    </ProfilePosts>
  </section>
</ProfileTabs>
`;
